core/src/main/java/pl/jutupe/core/repository/playlist/PlaylistLocalRepository.kt

```diff
@@ -0,0 +1,235 @@
+        MediaStore.Audio.Media.ARTIST,
+        MediaStore.Audio.Media.ALBUM,
+        MediaStore.Audio.Media.ALBUM_ID,
+    )
+
+    private val playlistProjection = arrayOf(
+        MediaStore.Audio.Playlists._ID,
+        MediaStore.Audio.Playlists.NAME,
+    )
+
+    private val playlistMemberProjection = arrayOf(
+        MediaStore.Audio.Playlists.Members._ID,
+        MediaStore.Audio.Playlists.Members.AUDIO_ID,
+    ) + songProjection
+
+    override suspend fun getAll(filter: Filter): List<MediaDescriptionCompat> {
+        Timber.d("getAll(filter=$filter)")
+
+        val cursor = context.contentResolver.queryPaged(
+            playlistsUri,
+            playlistProjection,
+            null,
+            null,
+            filter
+        )
+
+        return cursorToPlaylists(cursor!!)
+    }
+
+    override suspend fun findById(id: String): MediaDescriptionCompat? {
+        Timber.d("findById(mediaId=$id)")
+
+        val cursor = context.contentResolver.queryPaged(
+            playlistsUri,
+            playlistProjection,
+            "_id == ?",
+            arrayOf(id),
+            Filter()
+        )
+
+        return cursorToPlaylists(cursor!!).firstOrNull()
+    }
+
+    override suspend fun removeMembersByAudioId(playlistId: String, audioId: String) {
+        context.contentResolver.delete(
+            playlistMembersUri(playlistId),
+            MediaStore.Audio.Playlists.Members.AUDIO_ID + " =?",
+            arrayOf(audioId)
+        )
+    }
+
+    override suspend fun create(playlist: MediaDescriptionCompat): MediaDescriptionCompat {
+        Timber.d( "create(playlist=$playlist)")
+
+        val filter = Filter(Pagination.DEFAULT_PAGINATION)
+
+        getAll(filter).first().let {
+            it.mediaId?.let { it1 -> delete(it1) }
+        }
+
+        val playlistContentValue = ContentValues().apply {
+            put(MediaStore.Audio.Playlists._ID, playlist.mediaId)
+            put(MediaStore.Audio.Playlists.NAME, playlist.title.toString())
+            put(
+                MediaStore.Audio.Playlists.DATE_MODIFIED,
+                System.currentTimeMillis()
+            )
+        }
+
+        val createdUri = context.contentResolver.insert(playlistsUri, playlistContentValue)
+        Timber.d("created uri: $createdUri")
+
+        val cursor = context.contentResolver.queryPaged(
+            createdUri!!, playlistProjection, null, null, Filter()
+        )
+
+        return cursorToPlaylists(cursor!!).first()
+    }
+
+    override suspend fun delete(id: String) {
+        context.contentResolver.delete(
+            playlistsUri,
+            MediaStore.Audio.Playlists._ID + "=",
+            arrayOf(id))
+    }
+
+    override suspend fun getMembers(
+        playlistId: String,
+        filter: Filter
+    ): List<MediaDescriptionCompat>? {
+        Timber.d("getPlaylistMembers(playlistId=$playlistId, filter=$filter)")
+
+        //check if playlist exists
+        if (findById(playlistId) == null) {
```

Comment:
??

Reply:
We must check that playlist with given id exists, before fetching its members. :)